### ISSUE TITLE
HorizontalPodAutoscaler is using autoscaling/v2beta when autoscaling/v2 is out and in use

### ISSUE THREAD

**Issue by atropos112** [VictoriaMetrics/helm-charts#514]
As a person that is running on Kubernetes 1.26.3 I am having issues when enabling the helm-charts horizontalPodAutoscaler.

Namely it tells me I don't have appropriate CRD, this is because `autoscaling/v2beta2` is deprecated in k8s `1.23` and removed presumingly in 1.26? (not 100% sure, maybe it was 1.25). If backward compatibility is important here I guess adding option to override the version in the values.yaml override file might be an option? Either way using `autoscaling/v2beta2` will only become problematic to more people as time goes on and needs replacing sooner or later.

**Comment by b-a-t** [comment]
I got a Dejavu feeling reading this ticket and then recalled that I asked in Telegram:

> `error calculating structured merge diff: error calculating diff: error while running updater.Apply: converting (v2beta2.HorizontalPodAutoscaler) to (v2.HorizontalPodAutoscaler): unknown conversion`

> https://github.com/kubernetes/ingress-nginx/issues/8599

Indeed, I had to disable HPA for the `victoria-metrics-cluster` due to this error.

On another side, I'm not sure how HPA is useful here, as it affects only vminsert/vmselect pods, while you'd possibly be more interested in scaling up vmstorage nodes(I know it's not an easy task to perform).

Also, I was thinking of trying to replace plain HPA with the KEDA as a more flexible interface.
